### DISCUSSION TITLE
Is it possible to animate Clip geometry?

### DISCUSSION THREAD

**Discussion by dbriard** [AvaloniaUI/Avalonia#12690] [General]
Hi, 

I would like to reveal a control using a sliding rectangle geometry in clip, but Avalonia do not let me assign a name to the RectangleGeometry.
![image](https://github.com/AvaloniaUI/Avalonia/assets/3595443/597c2b9e-84a8-4299-a027-9d2fd490df5d)

In WPF, I am using a RectAnimation on the RectangleGeometry with -20,0,20,20 to 0,0,20,20.
Can I do something similar in Avalonia?

I am looking to do that:

https://github.com/AvaloniaUI/Avalonia/assets/3595443/b6fb3d4b-50e6-4b3a-a9f7-f607cfb99f47

Thanks

**Comment by timunie** [comment]
I guess you may need to provide your own transition for it to work. https://docs.avaloniaui.net/docs/next/guides/graphics-and-animation/transitions
https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Base/Animation/Transition.cs

**Reply by dbriard** [reply]
Hum, some kind of GeometryTransition, I'll give a try, thx.

**Reply by leonidcan** [reply]
Any progress on this? Are we going to have 'Clip' animations?
